fgpyo/sam/__init__.py

```diff
@@ -1203,6 +1227,12 @@ def set_tag(
         for rec in self.all_recs():
             rec.set_tag(tag, value)
 
+    def with_aux_alignments(self) -> "Template":
```

Comment:
```suggestion
    def with_aux_alignments(self, include_secondary: bool = True, include_supplementary: bool = False) -> "Template":
```

Are these sensible defaults?

By default, `bwa aln` will produce a _single_ record with all **secondary** alignments in the XA tag, and `bwa mem` will produce _multiple_ records, each with all other supplementary alignments in the SA tag

Alternatively (or in addition), we could confirm that rehydrating does not create any duplicate records on the template

Reply:
Yes! This refactor may actually help me with another design decision I didn't love but couldn't solve easily.

- https://github.com/fulcrumgenomics/fgpyo/pull/209#discussion_r1910662934